# HG changeset patch
# User Rik <rik@octave.org>
# Date 1423420436 28800
#      Sun Feb 08 10:33:56 2015 -0800
# Node ID f1270e5a31178f5a34fe836b0c3c17cee35cd71b
# Parent  c17e1cefdbd316d59815a13187fa3fafde7dafeb
__ghostscript__.m: Use del to remove files on Windows machines (bug #44186).

* __ghostscript__.m: Create new variable dos_shell which determines which
platform Octave is running on.  Use dos_shell var to decide whether to use 'rm'
or 'del' to remove files.

diff --git a/scripts/plot/util/private/__ghostscript__.m b/scripts/plot/util/private/__ghostscript__.m
--- a/scripts/plot/util/private/__ghostscript__.m
+++ b/scripts/plot/util/private/__ghostscript__.m
@@ -21,16 +21,18 @@
 ## Undocumented internal function.
 ## @end deftypefn
 
 ## Author: Ben Abbott <bpabbott@mac.com>
 ## Created: 2010-07-26
 
 function [gs_cmd, cleanup_cmd] = __ghostscript__ (varargin);
 
+  dos_shell = (ispc () && ! isunix ());
+
   opts.binary = "";
   opts.source = "-";
   opts.output = "-";
   opts.device = "";
   opts.epscrop = false;
   opts.antialiasing  = false;
   opts.antialiasing_textalphabits = 4;,
   opts.antialiasing_graphicsalphabits = 4;
@@ -108,17 +110,21 @@ function [gs_cmd, cleanup_cmd] = __ghost
     str = sprintf ("%s [%d %d] %s", "<< /Margins [0 0] /.HWMargins [0 0 0 0] /PageOffset",
                    opts.pageoffset, ">> setpagedevice");
     offset_ps = {"%!PS-Adobe-3.0", str, "%%EOF"};
     if (isfield (opts, "offsetfile"))
       offsetfile = opts.offsetfile;
       cleanup_cmd = "";
     else
       offsetfile = [tempname() ".ps"];
-      cleanup_cmd = sprintf ("rm %s", offsetfile);
+      if (dos_shell)
+        cleanup_cmd = ["del " strrep(offsetfile, '/', '\')];
+      else
+        cleanup_cmd = ["rm " offsetfile];
+      endif
     endif
     unwind_protect
       fid = fopen (offsetfile, "w");
       if (fid == -1)
         error ("print:fopenfailed", "__ghostscript__.m: fopen () failed");
       endif
       fprintf (fid, "%s\n", offset_ps{:});
     unwind_protect_cleanup
@@ -143,19 +149,28 @@ function [gs_cmd, cleanup_cmd] = __ghost
   endif
   if (! isempty (opts.prepend)
       && any (strcmpi (opts.device, {"pswrite", "ps2write", "pdfwrite"})))
     ## FIXME: Fonts get may be mangled when appending ps/ps2.
     ##        See "How to concatenate several PS files" at the link,
     ##        http://en.wikibooks.org/wiki/PostScript_FAQ
     cmd = sprintf ("%s %s", cmd, opts.prepend);
     if (isempty (cleanup_cmd))
-      cleanup_cmd = sprintf ("rm %s", opts.prepend);
+      if (dos_shell)
+        cleanup_cmd = ["del " strrep(opts.prepend, '/', '\')];
+      else
+        cleanup_cmd = ["rm " opts.prepend];
+      endif
     else
-      cleanup_cmd = sprintf ("%s ; rm %s", cleanup_cmd, opts.prepend);
+      if (dos_shell)
+        cleanup_cmd = sprintf ("%s & del %s", cleanup_cmd,
+                               strrep (opts.prepend, '/', '\'));
+      else
+        cleanup_cmd = sprintf ("%s ; rm %s", cleanup_cmd, opts.prepend);
+      endif
     endif
   endif
   if (! isempty (offsetfile) && format_for_printer)
     cmd = sprintf ("%s %s", cmd, offsetfile);
   endif
   if (! isempty (opts.source))
     cmd = sprintf ("%s %s", cmd, opts.source);
   endif
